### PR TITLE
Add rule interfaces and dummy implementations

### PR DESCRIPTION
**Pull request by duncanp-sonar**
Part of #3565

### CODE REVIEW

File: src/Rules/IRuleHelp.cs [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        /// </summary>
+        string RuleKey { get; }
+
+        Language Language { get; }
+
+        /// <summary>
+        /// The HTML des
+        /// </summary>
+        string HtmlDescription { get; }
+    }
```
Comment (duncanp-sonar) [inline src/Rules/IRuleHelp.cs:41]: Interface definitions with minimal class implementations and tests.

Comment (bigfluffycookie) [reply src/Rules/IRuleHelp.cs:41]: Sounds good!

File: src/Rules/IRuleMetadataProvider.cs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    [Export(typeof(IRuleMetadataProvider))]
+    [PartCreationPolicy(CreationPolicy.Shared)]
+    public class RuleMetadataProvider : IRuleMetadataProvider
+    {
+        public IRuleHelp GetRuleHelp(Language language, string ruleKey)
+        {
+            // TODO: fetch rule info
+            var htmlHelp = $@"Language: {language}, rule key: {ruleKey}";
+
+            return new RuleHelp(language, ruleKey, htmlHelp);
```
[the remark this answers is not in the document]
Comment (bigfluffycookie) [reply src/Rules/IRuleMetadataProvider.cs:46]: Yea fair enough

**Review by bigfluffycookie — APPROVED** [review]
LGTM!

**Pull request merged**